src/components/application_manager/test/commands/mobile/show_constant_tbt_test.cc

```diff
@@ -0,0 +1,395 @@
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <string>
+
+#include "application_manager/commands/mobile/show_constant_tbt_request.h"
+#include "application_manager/commands/command_request_test.h"
+#include "application_manager/policies/mock_policy_handler_interface.h"
+#include "application_manager/message_helper.h"
+#include "gtest/gtest.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace show_constant_tbt_request {
+
+namespace am = application_manager;
+using am::commands::ShowConstantTBTRequest;
+using ::utils::SharedPtr;
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+
+using policy_test::MockPolicyHandlerInterface;
+
+namespace {
+const uint32_t kAppId = 1u;
+const uint32_t kConnectionKey = 2u;
+const std::string kNavigationText1 = "Some text1";
+const std::string kNavigationText2 = "Some text2";
+const std::string kEta = "Some time";
+const std::string kTotalDistance = "Some distance";
+const std::string kTimeToDestination = "Some time";
+const std::string kTutnIconValue = "0x1A3F";
+const std::string kNextTutnIconValue = "0x1A40";
+}  // namespace
+
+class ShowConstantTBTRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  MessageSharedPtr CreateMessageWithParams() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    msg_params[am::strings::turn_icon][am::strings::value] = kTutnIconValue;
+    msg_params[am::strings::next_turn_icon][am::strings::value] =
+        kNextTutnIconValue;
+    msg_params[am::strings::navigation_text_1] = kNavigationText1;
+    msg_params[am::strings::navigation_text_2] = kNavigationText2;
+    msg_params[am::strings::eta] = kEta;
+    msg_params[am::strings::total_distance] = kTotalDistance;
+    msg_params[am::strings::time_to_destination] = kTimeToDestination;
+
+    smart_objects::SmartObject soft_buttons =
+        smart_objects::SmartObject(smart_objects::SmartType_Array);
+    soft_buttons[0] = "sds";
```

Comment:
@BSolonenko it is still not correct. `softButtons` key is an array of elements of type `SoftButton`, not strings as you expect.
You should either remove this assignment or create soft button element with a correct structure according to [MOBILE_API](https://github.com/smartdevicelink/sdl_core/blob/61c37f3763b7f84dca8f42ffdaa705d454d1b711/src/components/interfaces/MOBILE_API.xml#L904)

Reply:
@AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/1977/commits/3985fe7db835a397a23c5558a7a393de0d42011e